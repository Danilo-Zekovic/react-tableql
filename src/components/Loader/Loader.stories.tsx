import React from 'react'

import Loader from './Loader'
import { STORYBOOK_SECTIONS } from '../../../stories/constants'

export default {
  title: `${STORYBOOK_SECTIONS.buildingBlocks}|Loader`,
  component: Loader,
}

export const Default = () => <Loader />
