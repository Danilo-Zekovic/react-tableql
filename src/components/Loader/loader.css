.rtql-loader {
  border: 5px solid #f3f3f3; /* Light grey */
  border-top: 5px solid #3498db; /* Blue */
  border-bottom: 5px solid #3498db;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  animation: spin 1s linear infinite;
}

@keyframes spin {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}