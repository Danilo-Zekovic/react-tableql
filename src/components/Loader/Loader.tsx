import React from 'react'

import './loader.css'

const Loader = () => <div className="rtql-loader" />

export default Loader
