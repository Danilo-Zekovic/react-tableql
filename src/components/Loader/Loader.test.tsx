import React from 'react'
import { render } from '@testing-library/react'

import Loader from './Loader'

describe('<Loader>', () => {
  it('default', () => {
    const { container } = render(<Loader />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
