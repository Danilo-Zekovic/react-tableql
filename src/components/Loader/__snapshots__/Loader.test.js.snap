// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader> default 1`] = `
<div
  class="rtql-loader"
/>
`;
